presenters/thread.rb

```diff
@@ -70,27 +71,29 @@ def to_hash with_responses=false, resp_skip=0, resp_limit=nil
   #     children)
   #   response_count
   #     The total number of responses
-  def get_paged_merged_responses(thread_id, responses, skip, limit)
+  def get_paged_merged_responses(thread_id, responses, skip, limit, recursive=false)
     response_ids = responses.only(:_id).sort({"sk" => 1}).to_a.map{|doc| doc["_id"]}
     paged_response_ids = limit.nil? ? response_ids.drop(skip) : response_ids.drop(skip).take(limit)
     content = Comment.where(comment_thread_id: thread_id).
       or({:parent_id => {"$in" => paged_response_ids}}, {:id => {"$in" => paged_response_ids}}).
       sort({"sk" => 1})
-    {"responses" => merge_response_content(content), "response_count" => response_ids.length}
+    {"responses" => merge_response_content(content, recursive), "response_count" => response_ids.length}
```

Comment:
This will loop over all the responses, and discard children if recursive = false, right?  I think it would save time, memory, and bandwidth to *only* fill `content` with top-level responses for non-recursive queries.  

Might not even need to pass `recursive` down to `merge_response_content` if it's done here.